src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -421,6 +421,88 @@ private int GetMemoryContext(string memoryReference, int? offset, out IDebugMemo
             return hr;
         }
 
+        /// <summary>
+        /// Given 'expression', it will query the engine for an IDebugProperty2
+        /// </summary>
+        /// <param name="eb">Error handler to use in this method.</param>
+        /// <param name="expression">The expression to evaluate.</param>
+        /// <param name="frameId">Which frame to use and evaluate the expression on.</param>
+        /// <param name="isExecInConsole">If the current expression is from a console exec.</param>
+        /// <param name="flags">Flags used for EvaluateSync</param>
+        /// <param name="dapEvalFlags">EvaluationFlags used for DAPEvaluateSync</param>
+        /// <param name="property">The IDebugProperty2 of the 'expression'</param>
+        /// <exception cref="ProtocolException">In any step of the method that fails, it will throw a protocol execption using the ErrorBuilder.</exception>
+        private void GetDebugPropertyFromExpression(ErrorBuilder eb, string expression, int frameId, bool isExecInConsole, enum_EVALFLAGS flags, DAPEvalFlags dapEvalFlags, out IDebugProperty2 property)
+        {
+            property = null;
+
+            IDebugStackFrame2 frame;
+            bool success;
+            if (frameId == -1 && isExecInConsole)
```

Comment:
Do we treat SetExpression like isExecInConsole and can also pull from top frame?

Reply:
I don't think we need to. But up to you.